vtex/faststore
# Feat: Shipping Simulation Integration

## What's the purpose of this pull request?

This PR adds the `shipping` query, that brings the information that we need to the `ShippingSimulation` feature.

It also adds the `Address` Endpoint using the [GetAddressByPostalCode Rest API](https://developers.vtex.com/vtex-rest-api/reference/getaddressbypostalcode).

## How it works?

This is an example query that you can use inside some graphql playground tool:

```graphql
{
  shipping(
    items: [{ id: "99988211", quantity: 1, seller: "1" }]
    postalCode: "32808"
    country: "USA") {
      logisticsInfo {
        slas {
          carrier
          price
          shippingEstimate
          localizedEstimates
        }
      }
      address {
        city
        neighborhood
      }
    }
}
```

And the response could looks like:

```json
{
  "data": {
    "shipping": {
      "logisticsInfo": [
        {
          "slas": [
            {
              "carrier": "Normal",
              "price": 0,
              "shippingEstimate": "3bd",
              "localizedEstimates": "Up to 3 business days"
            }
          ]
        }
      ],
      "address": {
        "city": "ORLANDO",
        "neighborhood": null
      }
    }
  }
}
```

## How to test it?

You can either run this branch locally by running `yarn develop` inside `/packages/api` or check these fields already being used to render a ShippingSimulation in [this PR](https://github.com/vtex-sites/nextjs.store/pull/210).

You can also switch to another account like the `storecomponents` [here](https://github.com/vtex/faststore/blob/97256644e29fa9d5d29021ec62e26e7ed1e80fdf/packages/api/local/server.ts#L11) and play around with different products.

### Starters Deploy Preview

https://github.com/vtex-sites/nextjs.store/pull/210

## References

https://developers.vtex.com/vtex-rest-api/reference/cartsimulation
https://developers.vtex.com/vtex-rest-api/reference/getaddressbypostalcode
https://help.vtex.com/en/tutorial/logistics--53udnvI5eBy8DKo8FOjMoP
